test/es-module/test-cjs-esm-warn.js

```diff
@@ -10,66 +10,137 @@ const path = require('node:path');
 const { execPath } = require('node:process');
 const { describe, it } = require('node:test');
 
-
 const requiringCjsAsEsm = path.resolve(fixtures.path('/es-modules/cjs-esm.js'));
 const requiringEsm = path.resolve(fixtures.path('/es-modules/cjs-esm-esm.js'));
 const pjson = path.toNamespacedPath(
   fixtures.path('/es-modules/package-type-module/package.json')
 );
 
+describe(
+  'CJS ↔︎ ESM interop warnings',
+  { concurrency: !process.env.TEST_PARALLEL },
+  () => {
+    it(async () => {
+      const required = path.resolve(
+        fixtures.path('/es-modules/package-type-module/cjs.js')
+      );
+      const basename = 'cjs.js';
+      const { code, signal, stderr } = await spawnPromisified(execPath, [
+        '--no-experimental-require-module',
+        requiringCjsAsEsm,
+      ]);
+
+      assert.ok(
+        stderr
+          .replaceAll('\r', '')
+          .includes(
+            `Error [ERR_REQUIRE_ESM]: require() of ES Module ${required} from ${requiringCjsAsEsm} not supported.\n`
+          )
+      );
+      assert.ok(
```

Comment:
Can you either use `assert.match` or `spawnSyncAndAssert` (see https://github.com/nodejs/node/blob/99e0d0d2186065aa8ca9ab7cd94e1637bb166220/test/common/README.md#spawnsyncandexitcommand-args-spawnoptions-expectations )? Otherwise when mismatches happen it would be very difficult to see why without modifying the source and re-running it in e.g. the CI.